pkg/controller/datavolume/util.go

```diff
@@ -37,22 +39,46 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	"kubevirt.io/containerized-data-importer/pkg/common"
 	cc "kubevirt.io/containerized-data-importer/pkg/controller/common"
+	"kubevirt.io/containerized-data-importer/pkg/controller/populators"
+	featuregates "kubevirt.io/containerized-data-importer/pkg/feature-gates"
+	"kubevirt.io/containerized-data-importer/pkg/util"
 )
 
 const (
 	// AnnOwnedByDataVolume annotation has the owner DataVolume name
 	AnnOwnedByDataVolume = "cdi.kubevirt.io/ownedByDataVolume"
 
-	// MessageErrStorageClassNotFound provides a const to indicate the DV storage spec is missing accessMode and no storageClass to choose profile
-	MessageErrStorageClassNotFound = "DataVolume.storage spec is missing accessMode and no storageClass to choose profile"
+	// MessageErrStorageClassNotFound provides a const to indicate the PVC spec is missing accessMode and no storageClass to choose profile
+	MessageErrStorageClassNotFound = "PVC spec is missing accessMode and no storageClass to choose profile"
 )
 
 var (
-	// ErrStorageClassNotFound indicates the DV storage spec is missing accessMode and no storageClass to choose profile
+	// ErrStorageClassNotFound indicates the PVC spec is missing accessMode and no storageClass to choose profile
 	ErrStorageClassNotFound = errors.New(MessageErrStorageClassNotFound)
 )
 
+// RenderPvc renders the PVC according to StorageProfiles
+func RenderPvc(ctx context.Context, client client.Client, pvc *v1.PersistentVolumeClaim) error {
```

Comment:
What about setting virt default storage class if storageclass is unset?

Reply:
We get it for free from the call to `renderPvcSpecVolumeModeAndAccessModes()` which uses `GetStorageClassByNameWithVirtFallback()`.